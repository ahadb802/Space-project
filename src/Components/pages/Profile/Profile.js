import { useSelector } from 'react-redux';
import ListGroup from 'react-bootstrap/ListGroup';
import SingleUIProfile from '../../SingleUIProfile/SingleUIProfile';

function Profile() {
  const rocketList = useSelector((state) => state.rocketData.rockets);
  const missionsList = useSelector((state) => state.missions.missions);

  return (
    <section className="wrapper-profile">
      <div className="Rocket-Wrapper">
        <h2 className="Title">My Missions</h2>
        <ListGroup>
          {missionsList?.filter((mission) => mission.joined === true).map((mission) => (
            <SingleUIProfile
              key={mission.mission_id}
              id={mission.mission_id}
              name={mission.mission_name}
            />
          ))}
          {
            missionsList?.filter((mission) => mission.joined === true).length === 0
            && <ListGroup.Item>No reserved Mission</ListGroup.Item>
          }
        </ListGroup>
      </div>
      {/* Rocket Data */}
      <div className="Rocket-Wrapper">
        <h2 className="Title">My Rockets</h2>
        <ListGroup>
          {rocketList?.filter((rocket) => rocket.reserved === true).map((rocket) => (
            <SingleUIProfile key={rocket.id} id={rocket.id} name={rocket.name} />
          ))}
          {
            rocketList?.filter((rocket) => rocket.reserved === true).length === 0
            && <ListGroup.Item>No reserved Rockets</ListGroup.Item>
          }
        </ListGroup>
      </div>
    </section>
  );
}

export default Profile;
