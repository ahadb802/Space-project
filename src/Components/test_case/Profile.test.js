import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Profile from '../pages/profile/Profile';
import store from '../redux/configureStore';

require('@testing-library/jest-dom');

describe('Test for Profile component', () => {
  it('renders Profile component without any error', () => {
    const output = render(<Provider store={store}><Profile /></Provider>);
    expect(output).toMatchSnapshot();
  });
});
