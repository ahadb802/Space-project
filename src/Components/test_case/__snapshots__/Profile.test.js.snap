// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Profile component renders Profile component without any error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="wrapper-profile"
      >
        <div
          class="Rocket-Wrapper"
        >
          <h2
            class="Title"
          >
            My Missions
          </h2>
          <div
            class="list-group"
          >
            <div
              class="list-group-item"
            >
              No reserved Mission
            </div>
          </div>
        </div>
        <div
          class="Rocket-Wrapper"
        >
          <h2
            class="Title"
          >
            My Rockets
          </h2>
          <div
            class="list-group"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="wrapper-profile"
    >
      <div
        class="Rocket-Wrapper"
      >
        <h2
          class="Title"
        >
          My Missions
        </h2>
        <div
          class="list-group"
        >
          <div
            class="list-group-item"
          >
            No reserved Mission
          </div>
        </div>
      </div>
      <div
        class="Rocket-Wrapper"
      >
        <h2
          class="Title"
        >
          My Rockets
        </h2>
        <div
          class="list-group"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
